pkg/apis/postgres-operator.crunchydata.com/v1beta1/crunchy_bridgecluster_types.go

```diff
@@ -79,6 +88,21 @@ type CrunchyBridgeClusterSpec struct {
 	Storage resource.Quantity `json:"storage"`
 }
 
+type CrunchyBridgeClusterRoleSpec struct {
+	// The name of this PostgreSQL role. The value may contain only lowercase
+	// letters, numbers, and hyphen so that it fits into Kubernetes metadata.
+	// The above is problematic for us as Bridge has a role with an underscore.
+	// TODO: figure out underscore dilemma
+	// +kubebuilder:validation:Pattern=`^[A-Za-z][A-Za-z0-9\-_ ]*[A-Za-z0-9]$`
+	// +kubebuilder:validation:Type=string
+	Name string `json:"name"`
+
+	// The name of the Secret that will hold the role credentials.
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
+	// +kubebuilder:validation:Type=string
+	SecretName string `json:"secretName"`
```

Comment:
❓ Secret names are DNS subdomains. Is there a validation rule/shortcut for that?

🤔 We can start with a more limited rule (the one here) and expand it later.

Reply:
According to [this](https://github.com/CrunchyData/postgres-operator/blob/master/config/crd/bases/postgres-operator.crunchydata.com_crunchybridgeclusters.yaml#L232-L234), the regex we want is `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`

Reply:
I would think that the `kubebuilder:validation:Format` marker would accept an `IsDNS1123Subdomain` option given that the kubebuilder codebase has functions written specifically for validating this, but it doesn't seem to work and I can't find any documentation for what options they actually accept for the "Format" validation...

Reply:
AFAIK, `format` comes from OpenAPIv3. I see references to JSON Schema Validation elsewhere. I'm also not sure what Kubernetes has implemented (or not).

- https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation →
https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#schema-object →
https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#dataTypeFormat

- https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules →
https://swagger.io/specification/#data-types →
https://tools.ietf.org/html/draft-bhutton-json-schema-validation-00#section-7.3
  <br/>`hostname` looks close, but I'm not excited by following 3–5 RFCs down to the early internet. :notsure:

I see the same regex you pasted when I look [here](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation#IsDNS1123Subdomain). So that plus max-length sounds good to me.